libazureinit/src/imds.rs

```diff
@@ -206,4 +254,74 @@ mod tests {
             serde_json::from_value(os_profile);
         assert!(os_profile.is_err_and(|err| err.is_data()));
     }
+
+    // Runs a test around sending via imds::query() with a given statuscode.
+    async fn run_imds_query_retry(statuscode: &StatusCode) -> bool {
+        const IMDS_HTTP_TOTAL_TIMEOUT_SEC: u64 = 5 * 60;
+        const IMDS_HTTP_PERCLIENT_TIMEOUT_SEC: u64 = 30;
+        const IMDS_HTTP_RETRY_INTERVAL_SEC: u64 = 2;
+
+        let mut default_headers = header::HeaderMap::new();
+        let user_agent =
+            header::HeaderValue::from_str("azure-init test").unwrap();
+
+        // Run a local test server that replies with simple test data.
+        let serverlistener = TcpListener::bind("127.0.0.1:3000").await.unwrap();
```

Comment:
```suggestion
        let serverlistener = TcpListener::bind("127.0.0.1:0").await.unwrap();
```
will cause the OS to assign it a free port, and you've already written the rest of the test to use the result of `local_addr()` so it should Just Work.

Reply:
Done